internal/provider/resource_vdb_group.go

```diff
@@ -80,12 +99,116 @@ func resourceVdbGroupRead(ctx context.Context, d *schema.ResourceData, meta inte
 
 	d.Set("name", apiRes.GetName())
 	d.Set("vdb_ids", apiRes.GetVdbIds())
+	d.Set("tags", flattenTags(apiRes.GetTags()))
 	return diags
 }
 
 func resourceVdbGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*apiClient).client
+
+	vdbGroupId := d.Id()
+
+	if d.HasChange("name") || d.HasChange("vdb_ids") {
+		updateVdbGroupReq := *dctapi.NewUpdateVDBGroupParameters()
+		if d.HasChange("name") {
+			updateVdbGroupReq.SetName(d.Get("name").(string))
+		}
+		if d.HasChange("vdb_ids") {
+			updateVdbGroupReq.SetVdbIds(toStringArray(d.Get("vdb_ids")))
+		}
+
+		_, httpRes, err := client.VDBGroupsAPI.UpdateVdbGroupById(ctx, vdbGroupId).UpdateVDBGroupParameters(updateVdbGroupReq).Execute()
+		if diags := apiErrorResponseHelper(ctx, nil, httpRes, err); diags != nil {
+			return diags
+		}
+	}
+
+	if d.HasChange("tags") {
```

Comment:
This update logic looks too complex and I am not sure if we really want a lookup logic around. As of now there are no tag update api. we explicitly need to delete the old one and create a new one.
The logic here ultimately does the same api call and there are no data retrieval involved, i dont see lookup being advantageous. Rather we can simply remove the old tags and create the new ones if there are changes.
I may be overlooking something so would really want other's opinions on it.

Reply:
Since tags are used for managing access, there could be potential race conditions when tags that should not be deleted get deleted as part of the update if we take the approach of deleting and recreating all tags?